### PR TITLE
fix: SPLUNK_REALM env var taking precedence over endpoint

### PR DESCRIPTION
**Pull request by seemk**
:spiral_notepad: https://github.com/signalfx/splunk-otel-js/issues/640

Fixes the case where setting `SPLUNK_REALM` environment variable would take precedence over `endpoint` set programmatically. Programmatic API should always take precedence (as is the case with other options).

Setting the endpoint, whether via env vars or programmatically, now takes precedence over the realm config. If both are set, a warning is logged.

Misc:
* Added a fixed link to `opentelemetry-instrumentation-mongoose` package to satisfy link checker. This has been removed from Aspecto's repository. Will open another PR to use the OTel package.

### CODE REVIEW

**Review by t2t2 — APPROVED** [review]

**Comment by seemk** [comment]
The e2e test failures are unrelated to this PR, fixing it in another.

File: docs/advanced-config.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ This distribution supports all the configuration options supported by the compon
...
 | `OTEL_TRACES_SAMPLER_ARG`                                       |                         | Stable  | String value to be used as the sampler argument. Only be used if OTEL_TRACES_SAMPLER is set.
 | `SPLUNK_ACCESS_TOKEN`<br>`accessToken`                          |                         | Stable  | The optional access token for exporting signal data directly to SignalFx API.
-| `SPLUNK_REALM`<br>`realm`                                       |                         | Stable  | The name of your organization's realm, for example, ``us0``. When you set the realm, telemetry is sent directly to the ingest endpoint of Splunk Observability Cloud, bypassing the Splunk OpenTelemetry Collector.
+| `SPLUNK_REALM`<br>`realm`                                       |                         | Stable  | The name of your organization's realm, for example, ``us0``. When you set the realm, telemetry is sent directly to the ingest endpoint of Splunk Observability Cloud, bypassing the Splunk OpenTelemetry Collector. **NOTE**: when an explicit endpoint is set, the realm configuration is ignored.
```
[the remark this answers is not in the document]
Comment (theletterf) [reply docs/advanced-config.md:69]: Consider adding this sentences to the description of each `ENDPOINT` setting:

> Setting a value overrides the `SPLUNK_REALM` environment variable.

File: src/tracing/options.ts [hunk 2]
```diff
@@ -264,19 +264,18 @@ export function otlpSpanExporterFactory(options: Options): SpanExporter {
...
-    protocol = 'http/protobuf';
-
-    if (endpoint !== undefined) {
+    if (endpoint === undefined) {
+      endpoint = `https://ingest.${options.realm}.signalfx.com/v2/trace/otlp`;
+      protocol = 'http/protobuf';
+    } else {
       diag.warn(
-        'OTLP span exporter factory: explicit endpoint ignored due to realm being set.'
+        'OTLP span exporter factory: realm ignored due to explicit endpoint being set.'
```
Comment (theletterf) [inline src/tracing/options.ts:272]:
```suggestion
        'OTLP span exporter factory: Realm value ignored (full endpoint URL has been specified).'
```

File: test/options.test.ts [hunk 3]
```diff
@@ -359,24 +359,24 @@ describe('options', () => {
...
+      process.env.OTEL_EXPORTER_OTLP_TRACES_ENDPOINT = 'http://localhost:4317';
 
       const options = _setDefaultOptions();
       const exporters = options.spanExporterFactory(options);
       assert(Array.isArray(exporters));
       const [exporter] = exporters;
       sinon.assert.calledWith(
         logger.warn,
-        'OTLP span exporter factory: explicit endpoint ignored due to realm being set.'
+        'OTLP span exporter factory: realm ignored due to explicit endpoint being set.'
```
Comment (theletterf) [inline test/options.test.ts:373]:
```suggestion
        'OTLP span exporter factory: Realm value ignored (full endpoint URL has been specified).'
```

**Comment by seemk** [comment]
Thanks @theletterf

**Review by theletterf — APPROVED** [review]

**Pull request merged**